third_party/txt/src/txt/paragraph_builder_txt.cc

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "paragraph_builder_txt.h"
+#include "paragraph_txt.h"
+
+namespace txt {
+
+ParagraphBuilderTxt::ParagraphBuilderTxt(
+    const ParagraphStyle& style,
+    std::shared_ptr<FontCollection> font_collection)
+    : font_collection_(std::move(font_collection)) {
+  SetParagraphStyle(style);
+}
+
+ParagraphBuilderTxt::~ParagraphBuilderTxt() = default;
+
+void ParagraphBuilderTxt::SetParagraphStyle(const ParagraphStyle& style) {
+  paragraph_style_ = style;
+  paragraph_style_index_ = runs_.AddStyle(style.GetTextStyle());
+  runs_.StartRun(paragraph_style_index_, text_.size());
+}
+
+void ParagraphBuilderTxt::PushStyle(const TextStyle& style) {
+  size_t style_index = runs_.AddStyle(style);
+  style_stack_.push_back(style_index);
+  runs_.StartRun(style_index, text_.size());
+}
+
+void ParagraphBuilderTxt::Pop() {
+  if (style_stack_.empty())
```

Comment:
Google style requires curly braces.  From the looks of our code that isn't the rules for chromium?

Reply:
done